packages/client/src/components/Navigation/Navigation.component.js

```diff
@@ -1,30 +1,11 @@
-import React, { useEffect, useState, useCallback } from 'react';
-import { useAuthState } from 'react-firebase-hooks/auth';
-import { auth, db, logout } from '../../firebase';
-import { query, collection, getDocs, where } from 'firebase/firestore';
+import React from 'react';
 import { NavLink, Link } from 'react-router-dom';
-
+import { useUser } from '../../userContext';
 import './Navigation.styles.css';
 import { Button } from '../Button/Button.component';
 
 export const Navigation = () => {
-  const [user] = useAuthState(auth);
-  const [name, setName] = useState('');
-  const fetchUserName = useCallback(async () => {
-    try {
-      const q = query(collection(db, 'users'), where('uid', '==', user?.uid));
-      const doc = await getDocs(q);
-      const data = doc.docs[0].data();
-      setName(data.name);
-    } catch (err) {
-      /* error */
-    }
-  }, [user?.uid]);
-  useEffect(() => {
-    if (user) {
-      fetchUserName();
-    }
-  }, [user, fetchUserName]);
+  const { user, name, logout } = useUser();
```

Comment:
I think it would make more sense to name the custom hook `useUserContext`. Being aware that this is a context based value can be helpful.

Reply:
changed it in Navigation, Login, Signup components.